### ISSUE TITLE
mshell not installing on arm

### ISSUE THREAD

**Issue by Lukas-drz** [wavetermdev/waveterm#230]
**Describe the bug**
When connecting to a Raspberry Pi 4 I have the following error, it cannot install mshell on arm.

prompt> connecting to root@192.168.0.44...
prompt> error connecting to remote: invalid packet received from mshell client: raw[which: no mshell-v0.3 in (/usr/local/sbin:/usr/local/bin:/usr/bin:/root/.mshell)]
prompt> error reading from controlling-pty: read /dev/ptmx: input/output error
prompt> installing mshell v0.3.0 to root@192.168.0.44...
prompt> error, install failed: arch cannot be detected (might be incompatible with mshell): invalid uname machine type 'armv7l', mshell only supports aarch64 (amd64) and x86_64 (amd64)

**Desktop (please complete the following information):**
 - OS: Linux
 - Version 0.5.3-2

**Comment by sawka** [comment]
This could be tricky, as we've never tested a non-64-bit version of waveshell/mshell.  I just verified that it does build correctly, but I don't have a test environment.

We're close to shipping v0.6.0.  If you're up for it, I can build you a custom arm7 version of mshell that you could try manually to see if it works.  If it does (which is still a big if), I'd be happy to add it as a supported configuration for the following release.

**Comment by Lukas-drz** [comment]
Let's try it !
Give me the instructions to follow and I will send you the feedback :)

**Comment by esimkowitz** [comment]
@Lukas-drz What OS are you running? Is it the 32-bit Raspbian?

**Comment by Lukas-drz** [comment]
I'm using PiKVM OS on my raspberry pi.
It's based on arch linux.

**Comment by esimkowitz** [comment]
Yeah, seems like [PiKVM is only 32-bit](https://github.com/pikvm/pikvm/issues/711), that's probably why it isn't working. We make available an aarch64 build of mshell, but we don't currently bundle 32-bit versions. We can look into this, I think there was some concern around compatibility with some of the libraries we use...
